### PR TITLE
Drop support for Ember versions below v3.28, Drop support for Node versions below v14

### PR DESCRIPTION
**Pull request by rpemberton**
Update this addon to the ember-cli 4.9.2 blueprint using ember-cli-update

### CODE REVIEW

**Comment by rpemberton** [comment]
Anybody know why gh actions aren't running? They seem to be running on the [PR for my fork](https://github.com/rpemberton/ember-cli-app-version/pull/2).

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  try-scenarios:
+    name: ${{ matrix.try-scenario }}
+    runs-on: ubuntu-latest
+    needs: "test"
+    timeout-minutes: 10
+
+    strategy:
+      fail-fast: false
+      matrix:
+        try-scenario:
```
Comment (bertdeblock) [inline .github/workflows/ci.yml:59]: The addon still seems to support v3.16, so I think we should re-add `ember-lts-3.16`, `ember-lts-3.20` and `ember-lts-3.24` for the time being.

File: README.md [hunk 2]
```diff
@@ -4,24 +4,21 @@ ember-cli-app-version [![Build Status](https://travis-ci.org/ember-cli/ember-cli
...
 If you add build metadata to the version, this addon will automatically append SHA to the end of the version.
 
-Compatibility
-------------------------------------------------------------------------------
+## Compatibility
 
-* Ember.js v3.16 or above
-* Ember CLI v2.13 or above
-* Node.js v10 or above
+* Ember.js v3.28 or above
```
Comment (bertdeblock) [inline README.md:9]: Should remain v3.16 for now I think.

File: README.md [hunk 3]
```diff
@@ -4,24 +4,21 @@ ember-cli-app-version [![Build Status](https://travis-ci.org/ember-cli/ember-cli
...
+* Ember CLI v3.28 or above
```
Comment (bertdeblock) [inline README.md:10]: Same here.

File: README.md [hunk 4]
```diff
@@ -4,24 +4,21 @@ ember-cli-app-version [![Build Status](https://travis-ci.org/ember-cli/ember-cli
...
+* Node.js v14 or above
```
Comment (bertdeblock) [inline README.md:11]: Should remain v10 for now.

File: package.json [hunk 5]
```diff
@@ -20,59 +20,62 @@
...
+    "prettier": "^2.8.1",
+    "qunit": "^2.19.3",
+    "qunit-dom": "^2.0.0",
     "release-it": "^14.10.1",
-    "release-it-lerna-changelog": "^3.1.0"
+    "release-it-lerna-changelog": "^3.1.0",
+    "webpack": "^5.75.0"
+  },
+  "peerDependencies": {
+    "ember-source": "^3.28.0 || ^4.0.0"
```
Comment (bertdeblock) [inline package.json:75]:
```suggestion
    "ember-source": "^3.16.0 || ^4.0.0"
```

File: package.json [hunk 6]
```diff
@@ -20,59 +20,62 @@
...
   },
   "engines": {
-    "node": "10.* || >= 12"
+    "node": "14.* || 16.* || >= 18"
```
Comment (bertdeblock) [inline package.json:78]: This would be a breaking change, so best to do this in a separate PR.

**Review by bertdeblock — CHANGES_REQUESTED** [review]

File: package.json (hunk 6 above)
Comment (rpemberton) [reply package.json:78]: I was thinking this PR would be the breaking change PR

- Drop support for Node < 14
- Drop support for Ember < 3.28

[v5.0.0](https://github.com/ember-cli/ember-cli-app-version/releases/tag/v5.0.0) was done in a similar way

Comment (bertdeblock) [reply package.json:78]: Doing a blueprint update, dropping support for Ember versions and dropping support for Node versions are unrelated things, that's why I would've preferred doing these things in separate PRs. Then we would also have separate changelog entries pointing to separate PRs, which seems cleaner to me.

Comment (rpemberton) [reply package.json:78]: They seem related to me. My understanding is that blueprints are paired with specific Node and Ember versions, i.e they're part of the blueprint and were automatically updated by `ember-cli-update` in this case.

Comment (bertdeblock) [reply package.json:78]: A specific version of the blueprint indeed comes with a specific Ember and node support matrix, but that doesn't mean we must drop support for Ember and Node versions to do an internal blueprint update I feel. If we make separate / specific pull requests, it will be easier to generate a clear changelog.

Comment (NullVoxPopuli) [reply package.json:78]: have the ember-cli team considered not requiring specific sets of PRs for automated changelog generation?

it's kind of prohibitive to contributions when _both_ changes need/should happen anyway.

If we move to something like Changesets, we can have _much_ better changelogs:
Some examples:
 - https://github.com/CrowdStrike/ember-headless-table/blob/main/ember-headless-table/CHANGELOG.md
 - With a preview PR: https://github.com/NullVoxPopuli/ember-resources/pull/686 so folks can see what will be in the next release
 
 This moves the responsibility of managing the changelog from contributors to the repo owners, where it should be.
 
 Convention-based changelogs deter contributors, and I don't think we can afford that :sweat_smile:

Comment (NullVoxPopuli) [reply package.json:78]: :thinking: do you have thoughts on then forbidding ember-cli-update from running on addons? -- we could constrain it to do so. It seems "git history" and _the recommended way to update ember stuff_ are in conflict.

But regardless of it's the better thing to do, the work is done, and it's work we want in totality anyway, so... _why split it further_?

Consumers of this addon don't really care about the git history, and we're generally all familiar with blueprint updates.
